core/src/main/scala/cats/data/Ior.scala

```diff
@@ -794,9 +794,9 @@ sealed abstract class Ior[+A, +B] extends Product with Serializable {
 
   final def ===[AA >: A, BB >: B](that: AA Ior BB)(implicit AA: Eq[AA], BB: Eq[BB]): Boolean =
     fold(
```

Comment:
I tend to think a pattern matching based implementation would be both clearer and perform better.

Isn't it just four lines and pretty simple?

Reply:
Do you mean using a temporary tuple for matching, right?
```scala
(this, that) match {
  case (Ior.Left(a),    Ior.Left(aa))     => AA.eqv(a, aa)
  case (Ior.Right(b),   Ior.Right(bb))    => BB.eqv(b, bb)
  case (Ior.Both(a, b), Ior.Both(aa, bb)) => AA.eqv(a, aa) && BB.eqv(b, bb)
  case _ => false
}
```
because otherwise it will require a bit more that just four lines:
```scala
this match {
  case Ior.Left(a) =>
    that match {
      case Ior.Left(aa) => AA.eqv(a, aa)
      case _            => false
    }
  case Ior.Right(b) =>
    that match {
      case Ior.Right(bb) => BB.eqv(b, bb)
      case _             => false
    }
  case Ior.Both(a, b) =>
    that match {
      case Ior.Both(aa, bb) => AA.eqv(a, aa) && BB.eqv(b, bb)
      case _                => false
    }
}
```

Reply:
I did mean using a temporary tuple (which I would hope scala could inline away, but maybe it can't).

Reply:
I don't think Scala will inline it away, but I think the JIT hopefully would.

Reply:
done